internal-data-raw/create_target_data.R

```diff
@@ -82,14 +82,21 @@ get_expanded_tasks_outputs <- function(target_block) {
     names(output_meta),
     function(output_type) {
       output_type_meta <- output_meta[[output_type]]
-      data.frame(
-        output_type = output_type,
-        output_type_id = ifelse(
-          output_type == "sample",
-          NA,
-          output_type_meta$output_type_id |> unlist() |> unname()
+      if (output_type == "sample") {
+        return(
+          data.frame(
+            output_type = output_type,
+            output_type_id = NA
+          )
         )
-      )
+      } else {
+        return(
+          data.frame(
+            output_type = output_type,
+            output_type_id = output_type_meta$output_type_id |> unlist() |> unname()
+          )
+        )
+      }
```

Comment:
previous code tried to be clever with `ifelse`, but this was a bug: `ifelse` is vectorized, so ended up only keeping the first entry of `output_type_meta$output_type_id |> unlist() |> unname()` for the "else" case